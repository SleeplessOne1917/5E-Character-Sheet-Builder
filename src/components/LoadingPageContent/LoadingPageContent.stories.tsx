import { ComponentMeta, ComponentStory } from '@storybook/react';
import LoadingPageContent from './LoadingPageContent';

export default {
	title: 'Components/LoadingPageContent',
	component: LoadingPageContent,
	decorators: [
		Story => (
			<div style={{ height: '100vh', display: 'flex' }}>
				<Story />
			</div>
		)
	]
} as ComponentMeta<typeof LoadingPageContent>;

const Template: ComponentStory<typeof LoadingPageContent> = () => (
	<LoadingPageContent />
);

export const Default = Template.bind({});
