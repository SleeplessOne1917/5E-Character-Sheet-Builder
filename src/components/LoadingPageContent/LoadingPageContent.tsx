import { useEffect, useState } from 'react';
import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';
import MainContent from '../MainContent/MainContent';

const LoadingPageContent = () => {
	const [dots, setDots] = useState('');

	useEffect(() => {
		const dotsInterval = setInterval(() => {
			setDots(prev => {
				if (prev.length === 3) {
					return '';
				} else {
					return prev + '.';
				}
			});
		}, 1000);

		return () => {
			clearInterval(dotsInterval);
		};
	}, []);

	return (
		<MainContent testId="loading-page-content">
			<h1>Loading{dots}</h1>
			<div
				style={{
					display: 'flex',
					alignItems: 'center',
					justifyContent: 'center',
					height: '100%'
				}}
			>
				<LoadingSpinner />
			</div>
		</MainContent>
	);
};

export default LoadingPageContent;
