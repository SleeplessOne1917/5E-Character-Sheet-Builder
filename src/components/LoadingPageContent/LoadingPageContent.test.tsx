import { render, screen } from '@testing-library/react';
import * as stories from './LoadingPageContent.stories';
import { composeStories } from '@storybook/react';

const { Default } = composeStories(stories);

it('renders correctly', () => {
	render(<Default />);

	expect(screen.getByTestId('loading-page-content')).toMatchSnapshot();
});
