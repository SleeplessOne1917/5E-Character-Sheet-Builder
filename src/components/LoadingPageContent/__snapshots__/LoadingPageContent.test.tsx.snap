// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  class="main"
  data-testid="loading-page-content"
>
  <div
    class="content"
  >
    <h1>
      Loading
    </h1>
    <div
      style="display: flex; align-items: center; justify-content: center; height: 100%;"
    >
      <div
        class="container"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/Icons.svg#spinner"
          />
        </svg>
      </div>
    </div>
  </div>
</main>
`;
